src/module.c

```diff
@@ -2394,6 +2394,24 @@ void RM_Yield(RedisModuleCtx *ctx, int flags, const char *busy_reply) {
             /* Let redis process events */
             processEventsWhileBlocked();
         } else {
+            /* If it is not the main thread, we need to process events after the
+             * main thread enter acquiring GIL state in order to protect the ae and
+             * avoid potential race conditions. */
+            if (!pthread_equal(server.main_thread_id, pthread_self())) {
```

Comment:
```suggestion
            if ( 0 == pthread_equal(server.main_thread_id, pthread_self()) ) {
```
Please incorporate this for better readability. 

Also, please always follow this. 

Remember, Positive always overtakes negative (being optimistic). While you read as well as when you think in real life as well.

Reply:
Thanks, but I think using `!` is a very simple and readable way, at least it saves 2 characters.